src/components/ActivitiyForm/FormComponents/ImageUpload.tsx

```diff
@@ -0,0 +1,142 @@
+          0,
+          maxImages,
+        );
+        return Array.from(new Set(updatedUrls));
+      });
+    },
+    onError: (error) => {
+      console.error('이미지 업로드 실패:', error);
+    },
+  });
+
+  //이미지url이 변경되면 값을 업데이트
+  useEffect(() => {
+    if (maxImages === 1) {
+      setValue(name, imagePreviewUrls[0] || '');
+    } else {
+      setValue(name, imagePreviewUrls);
+    }
+  }, [imagePreviewUrls, name, setValue, maxImages]);
+
+  //선택된 파일을 업로드하고 최대 이미지 수 초과하지 않도록 관리
+  const handleImageChange = async (event: ChangeEvent<HTMLInputElement>) => {
+    const files = event.target.files;
+    if (files) {
+      const remainingSlots = maxImages - imagePreviewUrls.length;
+      const filesToUpload = Array.from(files).slice(0, remainingSlots);
+
+      try {
+        await Promise.all(
+          filesToUpload.map((file) => imageUploadMutation.mutateAsync(file)),
+        );
+      } catch (error) {
+        console.error('이미지 업로드 중 오류 발생:', error);
+      }
+    }
+  };
+
+  //이미지 삭제.
+  const handleDelete = (index: number) => {
+    const updatedUrls = imagePreviewUrls.filter((_, i) => i !== index);
+    setImagePreviewUrls(updatedUrls);
+  };
+
+  return (
+    <div className="p-4">
+      <div className="flex flex-wrap space-x-4">
+        <label
+          htmlFor={`image-upload-${label}`}
+          className={classNames(
+            'flex-center group h-206 w-206 flex-shrink-0 flex-col rounded-md border-2 border-dashed border-gray-300 focus:outline-none',
+            {
+              'opacity-50': imagePreviewUrls.length >= maxImages,
+              'hover:border-nomad-black': imagePreviewUrls.length < maxImages,
+            },
+          )}
+          style={{
+            cursor:
+              imagePreviewUrls.length >= maxImages ? 'not-allowed' : 'pointer',
+          }}
+          aria-label={label}
+        >
+          {imageUploadMutation.isPending ? (
+            <span>업로드 중...</span>
+          ) : (
+            <>
+              <PlusIcon
+                aria-label="등록 아이콘"
+                className="h-48 w-48 text-gray-400 group-hover:text-nomad-black"
+              />
+              <span className="mt-1 text-gray-400 group-hover:text-nomad-black">
+                {label}
+              </span>
+            </>
+          )}
+        </label>
+
+        <input
+          id={`image-upload-${label}`}
+          type="file"
+          onChange={handleImageChange}
+          accept="image/*"
+          className="hidden"
+          multiple={maxImages > 1}
+          disabled={
+            imagePreviewUrls.length >= maxImages ||
+            imageUploadMutation.isPending
+          }
+        />
+
+        {imagePreviewUrls.map((url, index) => (
+          <div key={index} className="relative p-4">
+            <Image
+              src={url}
+              alt="미리보기 이미지"
+              width={206}
+              height={206}
+              className="mr-14 rounded-lg object-cover"
+            />
```

Comment:
```suggestion
          <div key={index} className="relative size-[206px] p-4">
            <Image
              src={url}
              alt="미리보기 이미지"
              fill
              className="mr-14 rounded-lg object-cover"
            />
```

이렇게 하신다면 미리보기의 크기가 정사각형으로 유지될 것 같습니다!